tempesta_fw/http.c

```diff
@@ -608,15 +611,22 @@ tfw_http_req_cache_cb(TfwHttpReq *req, TfwHttpResp *resp)
 
 	if (resp) {
 		/*
-		 * The response is prepared, send it as is. The response
-		 * is either passed through from the back-end server, or
-		 * it is generated from the cache, so unrefer all its data.
+		 * The request is served from cache.
+		 * Send the response as is and unrefer its data.
 		 */
-		if (tfw_http_adjust_resp(resp, req) == 0)
-			tfw_cli_conn_send(req->conn, (TfwMsg *)resp, true);
+		if (tfw_http_adjust_resp(resp, req))
+			goto resp_err;
+		if (tfw_cli_conn_send(req->conn, (TfwMsg *)resp, true))
+			goto resp_err;
```

Comment:
It seems there is no sense to try to send client an error message if we couldn't send them normal message. When error message can be sent if normal message can't?